src/merkletree.ts

```diff
@@ -0,0 +1,174 @@
+import { BytesLike, HexString, compare } from './bytes';
+
+import {
+  MultiProof,
+  makeMerkleTree,
+  isValidMerkleTree,
+  getProof,
+  getMultiProof,
+  processProof,
+  processMultiProof,
+  renderMerkleTree,
+} from './core';
+
+import { MerkleTreeOptions, defaultOptions } from './options';
+import { validateArgument, invariant } from './utils/errors';
+
+export type MerkleTreeData<T extends Exclude<any, number>> = {
```

Comment:
Is there a reason why you went with `type` instead of `interface` here?

I don't see any concrete issues so feel free to ignore, but you can use `interface` here, and instead of using `&` in the more specific `*Data` types, you can use `extends`, which allows you to override fields like the `format` one with more narrow types in the same way. I think I prefer interfaces and extending, instead of intersections, whenever that's possible.